src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import className from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../TodosContext/TodosContext';
+
+type Props = {
+  item: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ item }) => {
+  const {
+    id,
+    title,
+    completed,
+  } = item;
+
+  const { setTodos } = useContext(TodosContext);
+  const [isEditable, setIsEditable] = useState(false);
+  const [editedTodo, setEditedTodo] = useState(title);
+
+  const fieldTitle = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (fieldTitle.current && isEditable) {
+      fieldTitle.current.focus();
+    }
+  }, [isEditable]);
+
+  const handleChangeStatus = () => {
+    setTodos(currentTodos => (
+      currentTodos.map(currentTodo => (
+        currentTodo.id === id
+          ? {
+            ...currentTodo,
+            completed: !currentTodo.completed,
+          }
+          : currentTodo
+      ))
+    ));
+  };
+
+  const handleRemoveTodo = () => {
+    setTodos(currentTodos => (
+      currentTodos.filter(curTodo => curTodo.id !== id)
+    ));
+  };
+
+  const handleChangeTitleTodo = (
+    event:React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setEditedTodo(event.target.value);
+  };
+
+  const saveChanges = () => {
+    setTodos(currentTodos => (
+      currentTodos.map(currTodo => (
+        currTodo.id === id
+          ? {
+            ...currTodo,
+            title: editedTodo,
+          }
+          : currTodo
+      ))
+    ));
+
+    setIsEditable(false);
+  };
+
+  const handleKeyboardAction = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setEditedTodo(title);
+      setIsEditable(false);
+    }
+
+    if (event.key === 'Enter' && editedTodo.trim().length) {
```

Comment:
`event.key === 'Enter` this condition is duplicated, so make sense to combine them